### ISSUE TITLE
SQL parse error for `t."b"`?

### ISSUE THREAD

**Issue by garyb** [quasar-analytics/quasar#2711]
As far as I understand it, `t."b"` should be treated the same as `t.b` - it seems like a necessity, since in theory maps can have EJson keys.

Query:
``` sql
SELECT t0."b" FROM `/spark-local/structure_test_hetero.json` AS t0
```

Error:
``` json
{
    "error": {
        "status": "Malformed SQL^2 query.",
        "detail": {
            "message": "operator '*' expected; but found `b'"
        }
    }
}
```

With unquoted b, it succeeds:
``` sql
SELECT t0.b FROM `/spark-local/structure_test_hetero.json` AS t0
```

**Comment by alissapajer** [comment]
We plan to support all of `EJson` as map keys, but currently we only support string keys. So in theory `t."b"` should work identically to `t.b`, where `t.b` could be thought of as a shorthand for `t."b"`. But the parser is just not yet prepared to handle this case. Similarly, you get the same parser error if you try `t.5` or `t.[1,2,3]`.

**Comment by jdegoes** [comment]
The preferred syntax for such things is `foo{[1,2,3]}` (project key `[1,2,3]` out of map `foo`), which hopefully is already supported by the parsers.

**Comment by garyb** [comment]
Yeah, I worked around it for now by making it "identifierify" string keys in the generated queries :)

The other cases won't happen anyway since this is generated by the structure editor, which means it'd only generate a query with other key literals if we actually had data with non-string keys in the first place. Just thought I'd open this in case it was unknown.
